tst/EnergyPlus/unit/ScheduleManager.unit.cc

```diff
@@ -1326,3 +1327,127 @@ TEST_F(EnergyPlusFixture, Schedule_GetCurrentScheduleValue_DST_RampUp_NoLeap)
+        "  1,                       !- Rows to Skip at Top",
+        "  8760,                    !- Number of Hours of Data",
+        "  Comma,                   !- Column Separator",
+        "  No,                      !- Interpolate to Timestep",
+        "  60,                      !- Minutes per item",
+        "  No;                     !- Adjust Schedule for Daylight Savings",
+        " ",
+        "Schedule:File,",
+        "  Test3,                   !- Name",
+        "  ,                        !- Schedule Type Limits Name",
+        "  " + scheduleFile.string() + ",              !- File Name",
+        "  2,                       !- Column Number",
+        "  1,                       !- Rows to Skip at Top",
+        "  8760,                    !- Number of Hours of Data",
+        "  Comma,                   !- Column Separator",
+        "  No,                      !- Interpolate to Timestep",
+        "  60,                      !- Minutes per item",
+        "  ;                     !- Adjust Schedule for Daylight Savings",
+        " ",
+        "Schedule:File,",
+        "  Test4,                   !- Name",
+        "  ,                        !- Schedule Type Limits Name",
+        "  " + scheduleFile.string() + ",              !- File Name",
+        "  2,                       !- Column Number",
+        "  1,                       !- Rows to Skip at Top",
+        "  8760,                    !- Number of Hours of Data",
+        "  Comma,                   !- Column Separator",
+        "  No,                      !- Interpolate to Timestep",
+        "  60;                      !- Minutes per item",
+        " ",
+    });
+
+    // This will process the provided idf chunk within the test fixture (must pass this step in order to proceed)
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    state->dataGlobal->NumOfTimeStepInHour = 1;
+    state->dataGlobal->MinutesPerTimeStep = 60;
+    state->dataGlobal->TimeStep = 1;         // Checking to see if omitting this is OK here
+    state->dataEnvrn->DayOfWeek = 1;         // Sunday
+    state->dataEnvrn->DayOfWeekTomorrow = 2; // Monday
+    state->dataEnvrn->DayOfYear_Schedule = 1;
+    state->dataGlobal->HourOfDay = 24;
+
+    // Test 1 condition
+    // "YES" Adjusts schedule for daylight savings
+    const int sch1idx = GetScheduleIndex(*state, "TEST1"); // Index of the IDF schedule object identified and stored into schdl1idx variable
+    ScheduleManager::ScheduleData &sch1 =
+        state->dataScheduleMgr->Schedule(sch1idx); // sch1 reference initialized that points to the specific schedule of interest
+    EXPECT_TRUE(sch1.UseDaylightSaving);           // Checks that the member variable got set correctly.
+
+    state->dataEnvrn->DSTIndicator = 1; // Tells the simulation that we're currently observing daylight savings
+    EXPECT_DOUBLE_EQ(ScheduleManager::LookUpScheduleValue(*state, sch1idx, state->dataGlobal->HourOfDay, state->dataGlobal->TimeStep), 0.0);
```

Comment:
Comparing the value from the CSV file that is looked up against the expected value of 0 for the 24th hour of the day.